example_external/CMakeLists.txt

```diff
@@ -0,0 +1,52 @@
+cmake_minimum_required(VERSION 3.16)
+project(generate_parameter_library_example_external)
+
+if(CMAKE_CXX_COMPILER_ID MATCHES "(GNU|Clang)")
+    add_compile_options(-Wall -Wextra -Wpedantic -Wshadow -Wconversion -Wsign-conversion -Wold-style-cast)
+endif()
+
+find_package(ament_cmake REQUIRED)
+find_package(ament_cmake_python REQUIRED)
+find_package(rclcpp REQUIRED)
+find_package(rclcpp_components REQUIRED)
+find_package(rclpy REQUIRED)
+find_package(generate_parameter_library_example REQUIRED)
+
+add_library(minimal_publisher_external SHARED
+  src/minimal_publisher_external.cpp
+)
+target_include_directories(minimal_publisher_external PUBLIC
+  $<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}/include>
+  $<INSTALL_INTERFACE:include/generate_parameter_library_example_external>
+)
+target_link_libraries(minimal_publisher_external
+  PUBLIC
+    rclcpp::rclcpp
+    rclcpp_components::component
+    generate_parameter_library_example::admittance_controller_parameters
```

Comment:
I'm just referring to the external example you provided: I did the same but not added it to the target_link_libraries, and it builds fine. (not sure about the implications with CMake and header-only libraries).

Reply:
If you comment out this line and try to build this package in a clean workspace, it should crash and say that it cannot find the parameters headers.

I haven't seen your example and the way dependencies are created and linked together, but my guess is that the generated parameter library is linked one way or another to your own library, preventing the crash.
Another guess is that since it is a header only library, so if you somehow indicate the headers' path in `target_include_directories`, the build won't crash.

Reply:
In other words, linking a target with an interface library comes down to adding the correct headers' path. This is the modern Cmake way of doing things, AFAIK.